src/python/pants/backend/project_info/dependencies.py

```diff
@@ -55,7 +60,10 @@ async def dependencies(
         dependencies_per_target_root = await MultiGet(
             Get(
                 Targets,
-                DependenciesRequest(tgt.get(DependenciesField), include_special_cased_deps=True),
+                DependenciesRequest(
+                    tgt.get(DependenciesField),
+                    should_resolve_deps_predicate=should_resolve_all_deps_predicate,
```

Comment:
Nit: could we rename this to `should_traverse_deps_predicate`? "resolve" is a very overloaded and vague term, and this is precisely about graph traversal.

It also helps clarify that we might still "resolve" (whatever that means) some dependency because we reached it through some other traversal path.

Reply:
I had `traverse` before when I only had this in `TransitiveTargetsRequest`. I changed to `resolve` when I pushed it into `DependenciesRequest` which is based on `resolve_dependencies`. Does `traverse` make sense with `DependenciesRequest`?

Reply:
I think we always want to "resolve" dependencies. We may or may not "traverse" the entire subtree though.

My 2c

Reply:
Renamed to use `traverse` instead of `resolve`.